### PR TITLE
Enhances detection of luajit

### PR DESCRIPTION
**Pull request by randall**
Enhances detection of luajit
    
If "--with-luajit" is specified, it will be used, otherwise configure will use pkg-config
or look in /usr and /usr/local

### CODE REVIEW

File: build/luajit.m4 [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      *":"*)
+        luajit_include="`echo $withval | sed -e 's/:.*$//'`"
+        luajit_ldflags="`echo $withval | sed -e 's/^.*://'`"
+        AC_MSG_CHECKING(checking for luajit includes in $luajit_include libs in $luajit_ldflags)
+        ;;
+      *)
+	# looking for versioned subdir
+        luajit_include="$withval/include"
+
+        AC_MSG_CHECKING(checking for $luajit_include/luajit-2.0 or $luajit_include/luajit-2.1)
```
Comment (shukitchan) [inline build/luajit.m4]: I think we don't yet support luajit>=2.1.x
Right?

**Comment by shukitchan** [comment]
I have been trying to do something similar as well. And I think some of the flags are not needed by the lua plugin. 

Also other things I was trying to do is to see if we can support lua as well as luajit and give more options to the community (now that we don't pull the luajit any more)

**Comment by shukitchan** [comment]
Also I mentioned "--enable-luajit" in the installation note for the lua plugin here - https://docs.trafficserver.apache.org/en/latest/admin-guide/plugins/lua.en.html#installation

We will want to update that as well.

File: build/luajit.m4 (hunk 1 above)
Comment (randall) [reply build/luajit.m4]: I don't know what our official line is, but I know @SolidWallOfCode was trying to build under FC28, which looks to be using luajit 2.1.x.

**Comment by randall** [comment]
@shukitchan I've updated the docs with this patch set to reference the new flag. 

I think support for regular Lua would be good as well, but I'd prefer that to be another change (though, it's probably mostly cut and paste with the luajit.m4)

Do you know which flags are not needed by the plugin?

File: build/luajit.m4 (hunk 1 above)
Comment (shukitchan) [reply build/luajit.m4]: ok... may be then we should be consistent with the pkg check ?

PKG_CHECK_MODULES([LUAJIT], [luajit < 2.1.0, luajit >= 2.0.4]

Right now it is <2.1.0

**Comment by randall** [comment]
The latest version of this change fixes up the include path, removing luajit from the global version and restricting them to just the plugin. 

And version restriction for pkg-config is reduced to >= 2.0.4

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Pull request merged**